File: src/main/webapp/js/Player.js
(function (window) {
    function Player(game, event, texture) {
        RemoteSprite.call(this, game, event, texture);

        this.shields = 3;
    }

    Player.prototype = Object.create(RemoteSprite.prototype);
    Player.prototype.constructor = Player;

    Player.prototype.shot = function shot(event) {
    };
    Player.prototype.hit = function hit(event) {
        this.shields--;
    };
    Player.prototype.update = function update(deltaTime) {
        Object.getPrototypeOf(Player.prototype).update.call(this, deltaTime);

        // this.residualVelocity = Math.max(this.residualVelocity - _playerDecelerationFactor, 0);
        // var currentVelocity = (this.forwardVelocity + this.residualVelocity);
        // var xVelocity = currentVelocity * Math.sin(this.rotation);
        // var yVelocity = currentVelocity * Math.cos(this.rotation);
        // this.x += xVelocity * _game.time.physicsElapsed;
        // this.y -= yVelocity * _game.time.physicsElapsed;
    };

    window.Player = Player;
})(window);

File: src/main/java/fr/kissy/zergling_push/Player.java
package fr.kissy.zergling_push;

import Event.PlayerMoved;
import io.netty.buffer.ByteBuf;

import java.nio.ByteBuffer;

/**
 * Created by Guillaume on 19/01/2017.
 */
public class Player {
    private static final double _playerVelocityFactor = 0.8;
    private static final double _playerAngularVelocityFactor = 0.006;
    private static final double _playerDecelerationFactor = 0.05;
    private static final double _moduloRadian = 2 * Math.PI;

    private float x;
    private float y;
    private double rotation;
    private byte velocity;
    private byte angularVelocity;
    private double residualVelocity;

    public Player() {
        this.x = 100;
        this.y = 100;
    }

    public void processMessage(ByteBuf message) {
        ByteBuffer byteBuffer = message.nioBuffer();
        if (PlayerMoved.PlayerMovedBufferHasIdentifier(byteBuffer)) {
            PlayerMoved event = PlayerMoved.getRootAsPlayerMoved(byteBuffer);
            this.x = event.y();
            this.y = event.y();
            this.rotation = event.rotation();
            this.velocity = event.velocity();
            this.angularVelocity = event.angularVelocity();
        }
    }

    public void update(long deltaTime) {
        this.residualVelocity = Math.max(this.residualVelocity - _playerDecelerationFactor, 0);

        double currentVelocity = (this.velocity + this.residualVelocity) * _playerVelocityFactor * deltaTime;
        this.x = clamp((float) (this.x + currentVelocity * Math.sin(this.rotation)), 0, 1920);
        this.y = clamp((float) (this.y - currentVelocity * Math.cos(this.rotation)), 0, 960);
        this.rotation = (this.rotation + this.angularVelocity * _playerAngularVelocityFactor * deltaTime) % _moduloRadian;
    }

    private float clamp(float value, float min, float max) {
        if (value < min) {
            return min;
        } else if (value > max) {
            return max;
        } else {
            return value;
        }
    }

    public float x() {
        return x;
    }

    public float y() {
        return y;
    }
}
